core/templates/services/extension-tag-assembler.service.ts

```diff
@@ -51,29 +51,32 @@ export class ExtensionTagAssemblerService {
     const traverseSchemaAndConvertSubtitledToDicts = (
         value: Object[] | Object
     ): Object[] | Object => {
-      let result: Object[] | Object;
+      let result!: Object[] | Object;
```

Comment:
Why the `!` here?

Reply:
Please take a look at [this discussion](https://github.com/oppia/oppia/pull/13369#discussion_r671870377)

Reply:
Again, please add a comment.

Reply:
done, please check if the comment is okay

Reply:
Sorry thought about this a bit and I think this whole function can be refactored
```typescript
const traverseSchemaAndConvertSubtitledToDicts = (
    value: Object[] | Object
): Object[] | Object => {
  if (value instanceof SubtitledUnicode || value instanceof SubtitledHtml) {     
    return value.toBackendDict();
  } else if (value instanceof Array) {
    return value.map(
      element => traverseSchemaAndConvertSubtitledToDicts(element));
  } else if (value instanceof Object) {
    type KeyOfValue = keyof typeof value;
    let _result: Record<KeyOfValue, Object> = {};
    result = {};
    let keys = <KeyOfValue[]>Object.keys(value);
    keys.forEach(key => {
      _result[key] = traverseSchemaAndConvertSubtitledToDicts(value[key]);
    });
    return _result as Object;
  }

  return value;
};
```

Reply:
done